src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +470,93 @@ public AgentSettings LoadSettings()
+                Trace.Info("AutoLogon will not be enabled as per user's input.");
+                return;
+            }
+
+            try
+            {
+                AssertAdminAccess();
+                
+                var iConfigManager = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+                iConfigManager.Configure(command);
+
+                if(iConfigManager.RestartNeeded())
+                {
+                    Trace.Info("AutoLogon is configured for a different user than the current user. Machine needs a restart.");
+                    _term.WriteLine(StringUtil.Loc("RestartMessage"));
+                    var shallRestart = command.GetRestartPermission();
+                    if(shallRestart)
+                    {
+                        Trace.Info("Restarting the machine in 5 seconds");
+                        _term.WriteLine(StringUtil.Loc("RestartIn5SecMessage"));
+                        Process.Start("shutdown.exe", "-r -t 5");
+                    }
+                    else
+                    {
+                        Trace.Info("No restart happened. As the interactive session is configured for a different user agent will not be launched");
+                    }
+                    return;
+                }
+
+                var startupProcessPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), "agentservice.exe");
+                Trace.Info("Launching the agent process.");
+                _term.WriteLine(StringUtil.Loc("AgentLaunch"));
+                
+                var processInfo = new ProcessStartInfo();
+                processInfo.UseShellExecute = false;
+                processInfo.CreateNoWindow = true;
+                processInfo.FileName = startupProcessPath;
+                processInfo.Arguments = "runasprocess";                
+
+                var agentServiceProcess = Process.Start(processInfo);
+                Trace.Verbose($"Started the agentservice process. Id - {agentServiceProcess.Id}");
+            }
+            catch(Exception ex)
+            {
+                Trace.Error(ex);
+                _term.WriteLine(StringUtil.Loc("AutoLogonConfigurationFailureMessage"));
+            }
+        }
+
+        private void UnConfigureAutoLogonIfNeeded()
+        {
+            AssertAdminAccess(true);
+            var iConfigManager = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+            if(!iConfigManager.IsInteractiveSessionConfigured())
+            {
+                Trace.Verbose("Interactive session was not configured on the agent. Returning.");
+                return;
+            }                
+            var listenerProcessId = GetAgentListenerProcessId();                
+            iConfigManager.UnConfigure(listenerProcessId);
+        }
+#endif
+
+        private int GetAgentListenerProcessId()
+        {
+            var processes = Process.GetProcessesByName("agent.listener");
```

Comment:
I have provided the reasons in UnConfigure() of InteractiveSessionConfigurationManager class. In our further sync ups I can explain as well.

---
In reply to: [117340566](https://github.com/Microsoft/vsts-agent/pull/978#discussion_r117340566) [](ancestors = 117340566)